### PR TITLE
Elasticsearch indexation

### PR DESCRIPTION
**Pull request by YanLabCHU**

### CODE REVIEW

File: src/main/java/bio/ferlab/clin/es/data/UpdateResponseListener.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public void onSuccess(Response response) {
+        try {
+            final String content = IOUtils.toString(response.getEntity().getContent(), Charset.defaultCharset());
+            final ElasticResponse.Index parsed = new ObjectMapper().readValue(content, ElasticResponse.Index.class);
+            if(parsed.getShards().getSuccessful() != 1 || parsed.getShards().getFailed() > 0){
+                throw new FailedToUpdateIndexException(this.index);
+            }
+        } catch (Exception e) {
```
Comment (jecos) [inline src/main/java/bio/ferlab/clin/es/data/UpdateResponseListener.java:29]: This exception is silently caught

File: src/main/java/bio/ferlab/clin/es/data/ElasticsearchData.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    public ElasticsearchData(RestClient client, String host, String index) {
+        this.client = client;
+        this.host = host;
+        this.index = index;
+    }
+
+    public void close() {
+        try {
+            this.client.close();
+        } catch (IOException e) {
```
Comment (jecos) [inline src/main/java/bio/ferlab/clin/es/data/ElasticsearchData.java:24]: This exception is silently caught

**Review by jecos — CHANGES_REQUESTED** [review]

File: src/main/java/bio/ferlab/clin/es/data/ElasticResponse.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        private String index;
+        private String type;
+        private String id;
+        private long version;
+        private String result;
+        private Shards shards;
+        private long seqNo;
+        private long primaryTerm;
+
+        @JsonProperty("_index")
```
Comment (plaplante-chust) [inline src/main/java/bio/ferlab/clin/es/data/ElasticResponse.java:16]: That's a personal preference but, I usually put the annotations on the field and use lombok for the getters/setters/constructors/hashcode/equals  etc.

File: src/main/java/bio/ferlab/clin/interceptors/IndexerInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Resource;
+import org.springframework.context.ApplicationContext;
+
+@Interceptor
+public class IndexerInterceptor {
+    private final ElasticsearchRestClient client;
+
+    public IndexerInterceptor(ApplicationContext appContext) {
+        // Autowired doesn't work, so we need the ApplicationContext to get the bean.
```
Comment (plaplante-chust) [inline src/main/java/bio/ferlab/clin/interceptors/IndexerInterceptor.java:20]: Did you try to implement ApplicationContextAware or use @Inject instead?

Comment (plaplante-chust) [inline src/main/java/bio/ferlab/clin/interceptors/IndexerInterceptor.java:20]: Did you try implementing ApplicationContextAware?

File: src/main/java/ca/uhn/fhir/jpa/starter/HapiProperties.java [hunk 5]
```diff
@@ -26,514 +26,538 @@
...
 
-  public static boolean isSSLValidationDisabled() {
-    return HapiProperties.getBooleanProperty(DISABLE_SSL_VALIDATION, false);
-  }
+    public static int getBioEsPort() {
+        return HapiProperties.getIntegerProperty(BIO_ES_PORT, 9200);
+    }
 
+    public static int getBioEsTransportPort() {
+        return HapiProperties.getIntegerProperty(BIO_ES_TRANSPORT_PORT, 9201);
```
Comment (plaplante-chust) [inline src/main/java/ca/uhn/fhir/jpa/starter/HapiProperties.java:556]: Isn't the default transport port usually 9300?

File: src/test/java/bio/ferlab/clin/es/ElasticsearchRestClientTest.java [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.*;
+
+public class ElasticsearchRestClientTest {
```
Comment (plaplante-chust) [inline src/test/java/bio/ferlab/clin/es/ElasticsearchRestClientTest.java:24]: :+1:

File: src/main/java/bio/ferlab/clin/interceptors/IndexerInterceptor.java (hunk 4 above)
Comment (YanLabCHU) [reply src/main/java/bio/ferlab/clin/interceptors/IndexerInterceptor.java:20]: Yes, I tried it and it didn't work

**Review by plaplante-chust — APPROVED** [review]

**Pull request merged**